pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -542,6 +542,11 @@ func parseAppInstanceConfig(getconfigCtx *getconfigContext,
 		appInstance.CloudInitVersion = cfgApp.CloudInitVersion
 		appInstance.FixedResources.CPUsPinned = cfgApp.Fixedresources.PinCpu
 
+		// Parse the snapshot related fields
+		if cfgApp.Snapshot != nil {
```

Comment:
Will it not be enough to initialize the SnapshotConfig struct instance with the default values?
In this case, it will create an instance of SnapshotConfig without explicitly setting any of its fields; the ActiveSnapshot field will be an empty string, MaxSnapshots will be 0, RollbackCmd will be nil, and Snapshots will be a nil slice.

Reply:
You're correct; I erroneously thought it started with the currently published AppInstanceConfg.